modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/ImplicitCastsTest.java

```diff
@@ -82,62 +82,30 @@ public void testNestedLoop(RelDataType lhs, RelDataType rhs, ExpectedTypes expec
     /** Filter clause - casts are added to condition operands. **/
     @ParameterizedTest
     @MethodSource("filterTypes")
-    public void testFilter(RelDataType lhs, ExpectedTypes expected) throws Exception {
+    public void testFilter(RelDataType lhs, RelDataType rhs, ExpectedTypes expected) throws Exception {
         IgniteSchema igniteSchema = new IgniteSchema("PUBLIC");
 
         addTable(igniteSchema, "A1", "COL1", lhs);
 
-        assertPlan("SELECT * FROM A1 WHERE COL1 > 1", igniteSchema, isInstanceOf(IgniteTableScan.class)
+        // Parameter types are not checked during the validation phase.
+        List<Object> params = List.of("anything");
+
+        assertPlan("SELECT * FROM A1 WHERE COL1 > CAST(? AS " + rhs + ")", igniteSchema, isInstanceOf(IgniteTableScan.class)
```

Comment:
`CAST(? AS " + rhs + ")"` -- this is not _implicit_ cast

Reply:
Implicit cast is added to the left side of the expression.